src/impl/realm_coordinator.cpp

```diff
@@ -796,6 +796,14 @@ void RealmCoordinator::on_change()
     for (auto& realm : m_weak_realm_notifiers) {
         realm.notify();
     }
+
+#if REALM_ENABLE_SYNC
+    // Invoke realm sync if another process has notified for a change
+    if (m_sync_session) {
+        auto versionId = m_db->start_read()->get_version_of_current_transaction();
```

Comment:
`m_db->get_version_of_latest_snapshot()` skips creating and destroying a read transaction

Reply:
thanks will fix

Reply:
@tgoyne This comment was bugging me. Probably I could fix this point as well ?
https://github.com/realm/realm-object-store/blob/d8ff6335be0872270489917edceeec5628e7a6fa/src/impl/realm_coordinator.cpp#L967

Reply:
That spot we need a VersionID rather than just the version number. I guess it could use get_version_of_latest_snapshot() for the check to see if it's a spurious wakeup and then only get the VersionID if it isn't.

Reply:
It seems that I could do it (the index seems not used anywhere after that point) but will need to construct a new VersionID only from the version_type which might introduce bugs later if the index is used somewhere from that semi VersionID

Reply:
yep that's good optimization

Reply:
`version` is eventually passed to `advance_read()`, which needs the index.

Reply:
I won't change that since I am worried changing that to non atomic read of the version.

Reply:
but I may be overthinking it